python-package/xgboost/sklearn.py

```diff
@@ -448,7 +448,7 @@ def fit(self, X, y, sample_weight=None, eval_set=None, eval_metric=None,
         else:
             evals = ()
 
-        self._features_count = X.shape[1]
+        self._features_count = len(X)
```

Comment:
I don't think this is the right way to go about fixing this issue. Perhaps XGBoost should guarantee that the input, `X`, is a numpy array when `fit` is called. Something simple like

```
X = np.array(X)
```